src/lib/y2firewall/clients/auto.rb

```diff
@@ -62,16 +90,30 @@ def change
         :next
       end
 
+      # Write the imported configuration to firewalld. If for some reason the
+      # configuration was not imported from the profile, it tries to import
+      # it again.
       def write
+        return false if !firewalld.installed?
+        import(self.class.profile) unless self.class.imported
```

Comment:
What about defining an `imported?` method? Just to keep consistence with `installed?`.